js/components/sharePanel.js

```diff
@@ -96,16 +96,23 @@ var SharePanel = React.createClass({
     mailToUrl += "?subject=" + encodeURIComponent(this.props.contentTree.title);
     mailToUrl += "&body=" + encodeURIComponent(emailBody + location.href);
     //location.href = mailToUrl; //same window
-    var emailWindow = window.open(mailToUrl, "email", "height=315,width=780"); //new window
-    setTimeout(function(){
-      try {
-         // If we can't access href, a web client has taken over and this will throw
-         // an exception, preventing the window from being closed.
-        var test = emailWindow.location.href;
-        emailWindow.close()
-      } catch(e) {};
-      // Generous 2 second timeout to give the window time to redirect if it's going to a web client
-    }, 2000);
+    if (OO.isIos && OO.isSafari) {
+        document.location = mailToUrl;
```

Comment:
Will this replace the currently loaded page?

Reply:
Yes. That's true. Previous page will be loaded with same states, once user send mail or cancel mail.

Reply:
@gramcha are there any side effects like the video being reloaded, etc?

Reply:
@jpfranco  no side effects. It will be restored to exact same state. I have tested it in iphone device multiple times.

Reply:
Ok, thanks for clarifying.